httpclient5/src/main/java/org/apache/hc/client5/http/impl/AuthSupport.java

```diff
@@ -54,20 +54,14 @@ public static void extractFromAuthority(
         if (authority == null) {
             return;
         }
-        final String userinfo = authority.getUserInfo();
-        if (userinfo == null) {
+        final String userInfo = authority.getUserInfo();
+        if (userInfo == null) {
             return;
         }
-        final int atColon = userinfo.indexOf(':');
-        final String userName;
-        final char[] password;
-        if (atColon >= 0) {
-            userName = userinfo.substring(0, atColon);
-            password = userinfo.substring(atColon + 1).toCharArray();
-        } else {
-            userName = userinfo.substring(0, atColon);
```

Comment:
I wonder how this could have worked before.

Reply:
@michael-o It is fairly unusual to have user info that does not conform to `username:password` format in HTTP.

Reply:
@michael-o On un match ':',  userName = userinfo.substring(0, atColon) will throw exception